runtime/v1/shim/service_linux.go

```diff
@@ -59,26 +63,85 @@ func (p *linuxPlatform) CopyConsole(ctx context.Context, console console.Console
 		}()
 	}
 
-	outw, err := fifo.OpenFifo(ctx, stdout, syscall.O_WRONLY, 0)
+	uri, err := url.Parse(stdout)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "unable to parse stdout uri")
 	}
-	outr, err := fifo.OpenFifo(ctx, stdout, syscall.O_RDONLY, 0)
-	if err != nil {
-		return nil, err
+
+	switch uri.Scheme {
+	case "binary":
+		ns, err := namespaces.NamespaceRequired(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		cmd := runtime.NewBinaryCmd(uri, id, ns)
+
+		// Create pipe to be used by logging binary for Stdout
+		outR, outW, err := os.Pipe()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create stdout pipes")
+		}
+
+		// Stderr is created for logging binary but unused when terminal is true
+		serrR, _, err := os.Pipe()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create stderr pipes")
+		}
+
+		r, w, err := os.Pipe()
+		if err != nil {
```

Comment:
Need to add cleanup to close `outR`, `serrR` and `r` if fail to start cmd

Reply:
Thanks, added the cleanup

Reply:
sorry for unclear comment. I think the cleanup should be called if there is any error after creation of pipe, like cmd.Start, w.Close() or r.Read(b).

Reply:
Thanks for clarifying, that makes sense. I've updated the commit.

Reply:
Is it possible to use defer to handle cleanup? Just thinking defer to reduce duplicate `runtime.CloseFiles(...)`.

Reply:
I believe so. Updated the commit to use defer.